js/src/common/components/ModalManager.js

```diff
@@ -33,7 +33,11 @@ export default class ModalManager extends Component {
   }
 
   animateShow(readyCallback) {
-    const dismissible = !!this.attrs.state.modal.componentClass.isDismissible;
+    const isDismissible = !!this.attrs.state.modal.componentClass.isDismissible;
+
+    // If the modal isn't dismissible, set these options to false.
+    const backdropDismissible = isDismissible ? !!this.attrs.state.modal.componentClass.doesBackdropClickDismiss : false;
```

Comment:
Couldn't these be `isDismissible && !!theAttr`?

Reply:
Yes, it could and it definitely should!

Reply:
I just realised that when `isDismissible` is true and backdrop dismissal is false, the modal will still close on backdrop click if using the `||` method as `isDismissible` will override the custom properties.

My intention was to only override if the dialog was not dismissible, otherwise we're in the same situation as before this PR.

Reply:
But wouldn't it work with `&&`?

Reply:
Yes... if only I didn't misread what you wrote 😅